### PR TITLE
fix: better support when options change dynamically

### PR DESCRIPTION
**Pull request by MatthieuLebigre**
## What I did

Added some logic to better handle the autocompletion when options changed (use case of a server interaction).

Fixes https://github.com/ing-bank/lion/issues/1395

### CODE REVIEW

**Comment by jorenbroekema** [comment]
Hey, thanks for making improvements to this, I actually was facing some non-ideal behavior as well when fetching options dynamically, not sure if your fix is fixing my issue too.

Can you elaborate a little bit on what this fixes, maybe a before & after GIF or a test that shows it?

**Comment by MatthieuLebigre** [comment]
More detailed on what I changed in this PR:

- When the option list changed (and after the user typed something in the textbox), the `_handleAutocompletion` is called again. Before my change, the logic in this method doesn't take into account that the user actually didn't changed anything, so a side effect is the `doAutoSelect()` is called, which leads to a very bad UX experience. I've changed the logic to take this into account and don't select any options if the user didn't trigger it.
- Also in `_handleAutocompletion()` method, for the current value if was taking as current value the whole textbox value, without removing the `selected` part of it, which again was causing trouble after the option list are changing
- Similirarly, I've added a check in `__textboxInlineComplete()` for not loosing inputNode.selectionStart and inputNode.selectionEnd

**Comment by MatthieuLebigre** [comment]
## Before

At the second letter entered, the autocomplete is too "aggressive" and autocomplete the word

lion-combobox-before

## After

Now it's working as intended:

lion-combobox-after

**Comment by jorenbroekema** [comment]
Alright makes sense I think. This was indeed the issue I was facing as well in my combobox implementation.

Would it be possible to create a regression test for this so we don't break this in the future?

**Comment by MatthieuLebigre** [comment]
> Alright makes sense I think. This was indeed the issue I was facing as well in my combobox implementation.
> 
> Would it be possible to create a regression test for this so we don't break this in the future?

done

**Comment by jorenbroekema** [comment]
I noticed there is a commit that shouldn't be in there for localize. This was a change done by Zubeyr, and it's already in master.

Maybe try rebasing / fixing the git problems.

Another thing, why a changeset to release all packages with a minor? We only need to release combobox right?

File: packages/combobox/src/LionCombobox.js [hunk 1]
```diff
@@ -629,7 +631,11 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
   _handleAutocompletion() {
     const hasSelection = this._inputNode.value.length !== this._inputNode.selectionStart;
 
-    const curValue = this._inputNode.value;
+    const inputValue = this._inputNode.value;
+    const inputSelectionStart = this._inputNode.selectionStart;
+    const curValue =
```
Comment (tlouisse) [inline packages/combobox/src/LionCombobox.js:636]: Nice. Makes sense to limit this variable just to the 'non selected part' 👍

File: packages/combobox/src/LionCombobox.js [hunk 2]
```diff
@@ -670,7 +676,10 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
           hasAutoFilled = true;
         };
 
-        if (userIntendsInlineAutoFill) {
+        if (
+          userIntendsInlineAutoFill ||
+          (this.__listboxContentChanged && this.__hadUserIntendsInlineAutoFill)
```
Comment (tlouisse) [inline packages/combobox/src/LionCombobox.js:681]: Maybe we can add this check inside `__computeUserIntendsAutoFill` method?

**Comment by tlouisse** [comment]
Very valuable contribution @MatthieuLebigre, thanks :)

It looks good, only had a small remark to move 1 line of code to a different method.

And then indeed the commit history (we only need the last one) and changeset need to be fixed as @jorenbroekema  pointed out

**Comment by MatthieuLebigre** [comment]
I fixed the latest comments, but didn't managed to restore properly the git history ... :(
I've opened a new one instead: https://github.com/ing-bank/lion/pull/1421
Sorry for the trouble...

**Pull request closed without merging**